jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaLayerConfigurations.java

```diff
@@ -110,6 +125,8 @@ public Builder setExplodedWarFiles(List<Path> explodedWarFiles) {
     }
 
     public JavaLayerConfigurations build() throws IOException {
+      Preconditions.checkState(appRoot.isAbsolute(), "'appRoot' must be an absolute path");
```

Comment:
this seems like we might need to use a custom validator that is not OS specific.

Reply:
Not sure I get what you mean. (Is `C:\dir` not absolute, for example? Even so, I think it is right to fail, since we assume we build a Linux image.)

Anyways, I think we have been assuming that this `appRoot` should always in Unix-style in the end.

Reply:
Whatever you meant, I think this will be taken care of properly after I `appRoot` from `Path` to `String`. The new condition will look like

```java
      // Windows filenames cannot have "/", so this also blocks Windows-style path.
      Preconditions.checkState(appRoot.startsWith("/"),
          "appRoot should be an absolute path in Unix-style");
```